modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/StackTraceProvider.scala

```diff
@@ -61,10 +59,11 @@ object StepFilterProvider {
       logger: Logger,
       testMode: Boolean
   ): StepFilterProvider = {
-    val scalaStepFilter = ScalaStepFilter(debuggee, tools, sourceLookUp, logger, testMode)
+    val scalaStepFilter : ScalaStepFilter = ScalaStepFilter(debuggee, tools, sourceLookUp, logger, testMode).asInstanceOf[ScalaStepFilter]
```

Comment:
Instead of casting to `ScalaStepFilter`, the `ScalaStepFilter.apply` method should return a `ScalaStepFilter`.

```suggestion
    val scalaStepFilter : ScalaStepFilter = ScalaStepFilter(debuggee, tools, sourceLookUp, logger, testMode)
```